### ISSUE TITLE
Learning kernel parameters: error dimensionality

### ISSUE THREAD

**Issue by LCanas** [cornellius-gp/gpytorch#1363]
Hi all, 

I just started using Gpytorch, and I was running a simple example to try to obtain a GP classifier for Nxf-dimensional input matrix.
In my case the kernel is fairly complex and due to the high number of samples I am also using an inducing kernel. I keep having this error:
The size of tensor a (20) must match the size of tensor b (17456) at non-singleton dimension 0

I also changed the example that I found in the documentation and I got a similar error:
import math
import torch
import gpytorch

from gpytorch.models import ApproximateGP
from gpytorch.variational import CholeskyVariationalDistribution
from gpytorch.variational import UnwhitenedVariationalStrategy

train_x = torch.rand(50, 2)
train_y = torch.sign(torch.cos(train_x * (4 * math.pi))).add(1).div(2)

class GPClassificationModel(ApproximateGP):
    def __init__(self, train_x):
        variational_distribution = CholeskyVariationalDistribution(train_x.size(0))
        variational_strategy = UnwhitenedVariationalStrategy(
            self, train_x, variational_distribution, learn_inducing_locations=False
        )
        super(GPClassificationModel, self).__init__(variational_strategy)
        self.mean_module = gpytorch.means.ConstantMean()
        self.covar_module = gpytorch.kernels.ScaleKernel(gpytorch.kernels.RBFKernel(ard_num_dims=2))

    def forward(self, x):
        mean_x = self.mean_module(x)
        covar_x = self.covar_module(x)
        latent_pred = gpytorch.distributions.MultivariateNormal(mean_x, covar_x)
        return latent_pred

# Initialize model and likelihood
model = GPClassificationModel(train_x)
likelihood = gpytorch.likelihoods.BernoulliLikelihood()

# this is for running the notebook in our testing framework
import os
smoke_test = ('CI' in os.environ)
training_iterations = 2 if smoke_test else 50

# Find optimal model hyperparameters
model.train()
likelihood.train()

# Use the adam optimizer
optimizer = torch.optim.Adam(model.parameters(), lr=0.1)

# "Loss" for GPs - the marginal log likelihood
# num_data refers to the number of training datapoints
mll = gpytorch.mlls.VariationalELBO(likelihood, model, train_y.numel())

for i in range(training_iterations):
    # Zero backpropped gradients from previous iteration
    optimizer.zero_grad()
    # Get predictive output
    output = model(train_x)
    # Calc loss and backprop gradients
    loss = -mll(output, train_y)
    loss.backward()
    print('Iter %d/%d - Loss: %.3f' % (i + 1, training_iterations, loss.item()))
    optimizer.step()

I keep having the error: The size of tensor a (50) must match the size of tensor b (2) at non-singleton dimension 1
Is there any dimensionality issue that I am not seeing?

Many thanks and sorry for the probable mistake.

**Comment by jacobrgardner** [comment]
Won't your train_y be 50x2 in the above code? I think you wanted to reduce over the last dimension somehow to get a length 50 vector, unless you explicitly wanted a multitask setup.

**Comment by moonskier** [comment]
I was havIng exactly the same issue (I tracked the error to an incompatibility between number of Gauss-Hermite quadrature points (the source of the value 20 in @LCanas's original post ) and the number of model output values (presumably the value 17456 in @LCanas's original post) , but didn't have a clue what was causing it.  Fortunately   @jacobrgardner's comment above pointed me to my mistake.  As in @LCanas's problem I'm fitting a GP classification model to nxd input data -- my mistake was making the response (y_train above) an torch tensor with shape (n,1) rather than (n,).
